enhancements/network/ovs-flow-export-configuration.md

```diff
@@ -0,0 +1,261 @@
+
+From the user-side configuration, the NOO has a custom resource definition that covers IPFix cache and sampling settings, deployment model (as a DaemonSet or as Deployment+Service) and a few other settings not relevant here.
+
+The NOO will generate a `ConfigMap` that is available for reading / watching to the CNO. The CNO would configure the OVN pods with the `ConfigMap` data.
+
+This solution ensures the NOO never interacts directly with OvS in order to avoid any configuration or state conflict with the CNO, and avoids creating a hard dependency in the CNO towards the NOO.
+
+Given the aforementioned details, following subsections depicts the work to do,
+grouped in different tasks.
+
+#### Task 1: setup a new ConfigMap that is readable by the CNO
+
+[Current `Network` operator configuration allows setting a static host:port address](
+https://docs.okd.io/latest/networking/ovn_kubernetes_network_provider/tracking-network-flows.html#nw-network-flows-object_tracking-network-flows).
+This information is forwarded to each OVN instance, on each cluster node. For example:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  exportNetworkFlows:
+    collectors:
+      - 192.168.1.99:2056
+```
+
+However, we need to set this address dynamically to ensure that each pod will forward
+the flows to the collector in its same node.
+
+While the above `Network` configuration is still valid for the current use cases, the NOO will generate a new `ConfigMap` that is exposed to the CNO, which
+will use it to configure the OVN pods. The CNO has to watch for any change in this configuration.
+
+See [diagrams](https://docs.google.com/document/d/1XxaFl4Gl1VbClfcPngkFCETv0anuFO6Klowh8MFZtJc/edit#) showing this interaction.
+
+The `ConfigMap` content sums up with any configuration that is already present in the Network CR. For example, if a user has configured an IP for netflow exports in the Network CR AND has configured a Netobs CR for the NOO, OvS will export to both destinations.
+
+Here is the new `ConfigMap` definition:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ovs-ipfix-config
+  namespace: # CNO's namespace
+spec:
+  sharedTarget: <service ip+port>
+  nodePort: <int>
+  sampling: <int>
+  cacheActiveTimeout: <int (seconds)>
+  cacheMaxFlows: <int>
+```
+
+Assuming that each OVS instance will forward flows to the same node (to minimize network traffic
+between nodes), the `nodePort` property will tell the CNO which port of its node should send data
+to. The node IP is discoverable by OVN pods or the CNO.
+
+`nodePort` and `sharedTarget` are mutually exclusive: the former will be set by the NOO when the collector is deployed as a DaemonSet, the latter will be set when deployed as Deployment + Service. So, when `sharedTarget` is used, the CNO will simply need to copy it to the ovn-k pods environment variable for IPFIX export that already exists.
```

Comment:
Forgive my ignorance here, but I thought OvS is running directly on a node and not within a pod. How can it export flows to a cluster service IP that is defined for sharedTarget?

Reply:
I'm not sure *how* it works internally but I can tell that it works since it's how I'm using it for a bunch of months now: deploying goflow with a service, grabbing its cluster IP, and setting it to `OVN_IPFIX_TARGETS` on `ovnkube-node` daemonset

Reply:
A process running on the node can still access the service IP address.  The host sends the service ip range in to ovn.